src/main/java/bio/terra/service/common/azure/StorageTableName.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.service.common.azure;
+
+import bio.terra.common.exception.NotImplementedException;
+import java.security.InvalidParameterException;
+import java.util.UUID;
+
+public enum StorageTableName {
+  SNAPSHOT_TABLE("snapshot"),
+  LOAD_HISTORY_TABLE("loadHistory"),
+  DEPENDENCIES_TABLE("dependencies"),
+  DATASET_TABLE("dataset") {
+    // TODO - With DR-2127, move remove special case for dataset
+    @Override
+    public String toTableName(UUID resourceId) {
+      throw new NotImplementedException(
+          "Dataset storage table names are not unique per resource. Use DATASET_TABLE.toTableName() instead.");
+    }
+
+    public String toTableName() {
+      return this.label;
+    }
+  },
+  FILES_TABLE("files") {
+    @Override
+    public String toTableName(UUID resourceId) {
+      throw new NotImplementedException(
+          "Files storage table names are not unique per resource. Use FILES_TABLE.toTableName() instead.");
+    }
+
+    public String toTableName() {
+      return this.label;
```

Comment:
Unnecessary `this.`. In general, we don't use `this.` in the jade codebase unless it's necessary for disambiguation, such as in a constructor or setter.
```suggestion
      return label;
```